test/functional/feature_assumeutxo.py

```diff
@@ -135,6 +135,19 @@ def expected_error(log_msg="", error_msg=""):
         rmtree(chainstate_snapshot_path)
         self.start_node(0)
 
+    def test_invalid_mempool_state(self, dump_output_path):
+        self.log.info("Test bitcoind should fail when mempool not empty.")
+        node=self.nodes[2]
+        tx = MiniWallet(node).send_self_transfer(from_node=node)
+
+        assert tx['txid'] in node.getrawmempool()
+
+        # Attempt to load the snapshot on Node 2 and expect it to fail
+        with node.assert_debug_log(expected_msgs=["[snapshot] can't activate a snapshot when mempool not empty"]):
+            assert_raises_rpc_error(-32603, "Unable to load UTXO snapshot", node.loadtxoutset, dump_output_path)
+
+        self.restart_node(2, extra_args=self.extra_args[2] )
```

Comment:
Please see my answer to your other question.

Reply:
There is an extra space after the extra_args option. 

Since the `-persistmempool=0` is only relevant to this one restart and not the rest of the tests I wouldn't change the setup and instead do it like this, but how it's done now is also ok.

Another small improvement might be to switch from node 2 to node 0. None of the args from node 2 are needed for this test, so using node 0, that has no args, may cause less confusion.

```suggestion
        # Restart to purge the transaction from the mempool again
        self.restart_node(0, extra_args=["-persistmempool=0"])
```

Reply:
Actually, when you switch to use node 0 you can remove this line entirely because node 0 is not used to load snapshots in the rest of the test.